### PR TITLE
Listen to browser track changes

### PR DESCRIPTION
**Pull request by egreaves**
### Changes proposed in this pull request:
Listen to browser track changes. Ignore tracks that aren't in use.
Fixes #
JW7-2687

### CODE REVIEW

File: src/js/providers/tracks-mixin.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ define(['../utils/underscore',
 
         if (this._textTracks) {
             if (this._renderNatively) {
-                this.addTracksListener(this._textTracks, 'change', textTrackChangeHandler.bind(this));
+                this.addTracksListener(this.video.textTracks, 'change', textTrackChangeHandler.bind(this));
```
Comment (pajong) [inline src/js/providers/tracks-mixin.js]: This is somewhat confusing - we keep ```this._textTracks``` and check if that exists, and use ```this.video.textTracks``` to add listener to. It feels like we are unnecessarily keeping two copies of the same thing.

File: src/js/providers/tracks-mixin.js [hunk 2]
```diff
@@ -300,20 +300,27 @@ define(['../utils/underscore',
...
-
-        if (!this._textTracks || this.video.textTracks.length > this._textTracks.length) {
+        var textTracks = this.video.textTracks;
+        var inUseTracks = _.filter(textTracks, function (track)  {
+            return track.inuse || !track._id;
+        });
+        if (!this._textTracks || inUseTracks.length > this._textTracks.length) {
             // If the video element has more tracks than we have internally..
-            this.setTextTracks.call(this, this.video.textTracks);
+            this.setTextTracks.call(this, textTracks);
```
Comment (pajong) [inline src/js/providers/tracks-mixin.js]: Again, we use ```textTracks``` and ```this._textTracks``` - feels redundant.

File: src/js/providers/tracks-mixin.js [hunk 3]
```diff
@@ -300,20 +300,27 @@ define(['../utils/underscore',
...
         var _selectedTextTrackIndex = -1, i = 0;
         for (i; i < this._textTracks.length; i++) {
             if (this._textTracks[i].mode === 'showing') {
                 _selectedTextTrackIndex = i;
                 break;
             }
         }
-        this.setSubtitlesTrack.call(this, _selectedTextTrackIndex + 1);
+        // Notifying the model when the index changes keeps the current index in sync in iOS Fullscreen mode
+        if(_selectedTextTrackIndex !== this._currentTextTrackIndex) {
```
Comment (pajong) [inline src/js/providers/tracks-mixin.js]: Minor - space after ```if```

File: src/js/providers/tracks-mixin.js [hunk 4]
```diff
@@ -300,20 +300,27 @@ define(['../utils/underscore',
...
+            console.log('new track index:',_selectedTextTrackIndex, this._currentTextTrackIndex);
```
Comment (pajong) [inline src/js/providers/tracks-mixin.js]: We should remove the console log.

File: src/js/providers/tracks-mixin.js (hunk 1 above)
Comment (hghazzi) [inline src/js/providers/tracks-mixin.js]: I agree that we should eventually find a way to consolidate the video tag tracks and _textTracks to be one and the same, or group them in a way to get what we need when we need it.

**Pull request merged**